### PR TITLE
Adds a festivizer to christmas that paints things red or green.

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

# Explain why it's good for the game

> Adds a festivizer to christmas that paints things red or green. Usable by both humans and xenomorphs.

It lets marines AND xenos spread the holiday cheer everywhere!

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
admin: Adds a festivizer to christmas that paints things red or green. Usable by both humans and xenomorphs.
/:cl:

### CODE REVIEW

**Comment by Git-Nivrak** [comment]
Literally 1984

File: code/game/objects/items/toys/toys.dm [hunk 1]
```diff
@@ -504,3 +504,30 @@
...
+	icon_state = "barbed_wire"
+
+/obj/item/toy/festivizer/examine(mob/user)
+	. = ..()
+	to_chat(user, SPAN_BOLDNOTICE("You see another label on \the [src] that says: <i> INCLUDES SUPPORT FOR FOREIGN BIOFORMS! </i> You're not sure you like the sound of that.."))
+
+/obj/item/toy/festivizer/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(target.color)
+		to_chat(user, SPAN_NOTICE("Tthe [target] is already colored, don't be greedy!"))
```
Comment (stanalbatross) [inline code/game/objects/items/toys/toys.dm:521]: Tthe

File: code/game/objects/items/toys/toys.dm [hunk 2]
```diff
@@ -504,3 +504,30 @@
...
+
+/obj/item/toy/festivizer
+	name = "\improper C92 pattern 'Festivizer' decorator"
+	desc = "State of the art, WY-brand, high tech... ah who are we kidding, it's just a festivizer. You spot a label on it that says: <i> Attention: This device does not cover item in festive wire, but rather paints it a festive color. </i> What a rip!"
+	icon = 'icons/obj/items/marine-items_christmas.dmi'
+	icon_state = "barbed_wire"
+
+/obj/item/toy/festivizer/examine(mob/user)
+	. = ..()
+	to_chat(user, SPAN_BOLDNOTICE("You see another label on \the [src] that says: <i> INCLUDES SUPPORT FOR FOREIGN BIOFORMS! </i> You're not sure you like the sound of that.."))
```
Comment (stanalbatross) [inline code/game/objects/items/toys/toys.dm:516]: use get examine text

File: code/game/objects/items/toys/toys.dm [hunk 3]
```diff
@@ -504,3 +504,30 @@
...
+
+/obj/item/toy/festivizer/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(target.color)
+		to_chat(user, SPAN_NOTICE("Tthe [target] is already colored, don't be greedy!"))
+		return
+	if(!target.Adjacent(user))
```
Comment (stanalbatross) [inline code/game/objects/items/toys/toys.dm:523]: This should be first

File: code/game/objects/items/toys/toys.dm [hunk 4]
```diff
@@ -504,3 +504,30 @@
...
+		return
+	target.color = (pick("red", "green"))
+	target.visible_message(SPAN_GREEN("\The [target] has been festivized by [user]! Merry christmas!"))
```
Comment (stanalbatross) [inline code/game/objects/items/toys/toys.dm:526]:
```suggestion
	target.visible_message(SPAN_GREEN("\The [target] has been festivized by [user]! Merry Christmas!"))
```

File: code/game/objects/items/toys/toys.dm [hunk 5]
```diff
@@ -504,3 +504,30 @@
...
+	to_chat(user, SPAN_GREEN("You festivize \the [target]! Merry christmas!"))
```
Comment (stanalbatross) [inline code/game/objects/items/toys/toys.dm:527]:
```suggestion
	to_chat(user, SPAN_GREEN("You festivize \the [target]! Merry Christmas!"))
```

File: code/game/objects/items/toys/toys.dm [hunk 6]
```diff
@@ -504,3 +504,30 @@
 							/obj/item/toy/prize/odysseus					= 1,
 							/obj/item/toy/prize/phazon						= 1
 							)
+
+/obj/item/toy/festivizer
```
Comment (stanalbatross) [inline code/game/objects/items/toys/toys.dm:508]: Where's the can be held by xenos flag

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
Why can't I draft on mobile

File: code/game/objects/items/toys/toys.dm (hunk 6 above)
Comment (carlarctg) [reply code/game/objects/items/toys/toys.dm:508]: there isn't one but thats what this proc is for

image

File: code/game/objects/items/toys/toys.dm (hunk 5 above)
Comment (harryob) [reply code/game/objects/items/toys/toys.dm:527]: u didn't do this carlarc

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys/toys.dm (hunk 5 above)
Comment (carlarctg) [reply code/game/objects/items/toys/toys.dm:527]: i did i just didn't use the suggestion... i did it manually.... stfu!

**Comment by carlarctg** [comment]
WHAT

**Review by harryob — APPROVED** [review]

**Review by stanalbatross — APPROVED** [review]
harry approved so i will too

**Pull request merged**